public/js/dashboard.js

```diff
@@ -61,50 +61,61 @@ requirejs([
         $container.removeClass('invisible');
     }
 
+    // delegates for kernel message handling, keyed by message type
+    // all functions receive the entire response message and a reference to an
+    // outputArea model associated with the code / widget that made the initial
+    // request associated with the received response
```

Comment:
Nit: This seems like multiple sentences. Add caps and punctuation to make it more readable.